bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+the following form:
+
+```
+OP_1 <taproot_output_key> 
+```
+
+where 
+```
+taproot_internal_key = MuSig2.KeyAgg(MuSig2.KeySort(bitcoin_key_1, bitcoin_key_2))
+
+taproot_output_key = taproot_internal_key + tagged_hash("TapTweak", taproot_internal_key) * G
+```
+
+`taproot_interal_key` is the aggregate key of `bitcoin_key_1` and
+`bitcoin_key_2` after first sorting the keys using the
+[MuSig2 KeySort][musig-keysort] algorithm and then running the
+[MuSig2 KeyAgg][musig-keyagg] algorithm.
+
+Then, to commit to spending the taproot output via the keyspend path, a
+[BIP86][bip86-tweak] tweak is added to the internal key to calculate the
+`taproot_output_key`.
+
+As with legacy channels, nodes will want to perform some checks before adding 
+an announced taproot channel to their routing graph:
+
+1. The funding output of the channel being advertised exists on-chain, is an
+   unspent UTXO and has a sufficient number of confirmations. To allow this 
+   check, the SCID of the channel will continue to be included in the channel
+   announcement.
+2. The funding transaction is sufficiently bound to the Lightning context. 
+   Nodes will do this by checking that they are able to derive the output key 
+   found on-chain by using the advertised `bitcoin_key_1` and `bitcoin_key_2` 
+   along with a [BIP86 tweak][bip86-tweak]. This provides a slightly weaker
+   binding to the LN context than legacy channels do but at least somewhat 
+   limits how the output can be spent due to the script-path being disabled.
+   The context binding with legacy channels is greater because an attacker 
+   trying to create gossip spam would need to create 2-of-2 multisig P2WSH 
+   outputs on-chain that would require them to then pay for the bytes of two
+   signatures at the time of spending the output. This extra cost provided some
+   extra deterrence for attackers. This deterrence is not present with P2TR 
+   transactions.
+3. The owners of `bitcoin_key_1` and `bitcoin_key_2` agree to be associated with  
+   a channel owned by `node_1` and `node_2`.
+4. The owners of `node_ID_1` and `node_ID_2` agree to being associated with the 
+   output paying to `bitcoin_key_1` and `bitcoin_key_2`. 
+
+For legacy channels, these last two proofs are made possible by including four
+separate signatures in the `channel_announcement`. However, [BIP-340][bip-340]
+signatures and the MuSig2 protocol allow us to now aggregate these four
+signatures into a single one. Verifiers will then be able to aggregate the four
+keys (`bitcoin_key_1`, `bitcoin_key_2`, `node_ID_1` and `node_ID_2`) using
+MuSig2 key aggregation and then they can do a single signature verification
+check instead of four individual checks.
+
+## Timestamp fields
+
+In this document, the `timestamp` fields in messages are block heights instead
+of the UNIX timestamps used in the legacy gossip messages. 
+
+### Rate Limiting
+
+A block height based timestamp results in more natural rate limiting for gossip
```

Comment:
Pretty cool how we can indirectly gain global rate limiting by using block hashes. I think this has a few implications for the ways `channel_updates` work in the network today:
  * Any attempts to use `max_htlc` to leak out the available bandwidth for a channel (so increase/decrease with each forward) can be rate limited at a global level. 
  * Protocols that would directly require a node to trigger a specific channel update are now less attractive from an incentives PoV as: those updates would eat into the node's available set of global updates within a block interval. 
  * Nodes can _still_ do more rapid updates by sending their new `channel_update` in an onion error _directly_ to the sender. 

Also some implementations have extensions to the gossip query protocol to support timestamp channel update syncing, which can now be updated to use block height based syncing instead.